# ML (sklearn) with UGA Covid dataset

In [5]:
# Classification with various learners and various dataset

from Library.Import import *
from Library.Utilities import svm_classifier, MLP, XGB, GP 
from Library.Utilities import bayes_classifier, decision_tree_classifier
from Library.Utilities import LeaveXout, read_XY
from sklearn.metrics import r2_score, accuracy_score, f1_score, matthews_corrcoef

DIRECTORY = './Dataset_input/Covid/UGA/'
dataset = 'MEDIUM' #  ALL or ALL_selected_features or MEDIUM
filename = f'{DIRECTORY}{dataset}'

seed = 1
np.random.seed(seed=seed)
niter = 3
selection = 1 # if > 0 remove feature
verbose = False

scoring_function = accuracy_score
feature, X, y = read_XY(filename, scaling='XY')
xfold = 20

for learner in [XGB]:

    acc_avr, acc_dev, F = LeaveXout(X, y.ravel(), feature[:-1], 
                                      learner=learner, scoring_function=scoring_function, 
                                      xfold=xfold, niter=niter, 
                                      selection=selection, verbose=True)

    print(f'{dataset} Size: {len(F)} Method: {learner.__name__} Score: {scoring_function.__name__} '
          f'Acc: {acc_avr:.3f}±{acc_dev:.3f} '
          f'for {xfold}-fold-CV and {niter} iter')

    if selection:
       F = np.array2string(F).replace('[','').replace(']','')
       print(f'Selected features:\n{F}')


Size: 55 Remove: EX_adn_e_i Score: 0.724±0.006
Size: 54 Remove: EX_ddca_e_i Score: 0.737±0.006
Size: 53 Remove: EX_thym_e_i Score: 0.749±0.015
Size: 52 Remove: EX_arab__L_e_i Score: 0.753±0.020
Size: 51 Remove: EX_acgal_e_i Score: 0.753±0.027
Size: 50 Remove: EX_ala__L_e_i Score: 0.761±0.025
Size: 49 Remove: EX_tyr__L_e_i Score: 0.770±0.029
Size: 48 Remove: EX_glu__L_e_i Score: 0.770±0.015
Size: 47 Remove: EX_tyrp_e_i Score: 0.774±0.015
Size: 46 Remove: EX_orn_e_i Score: 0.778±0.010
Size: 45 Remove: EX_maltttr_e_i Score: 0.778±0.027
Size: 44 Remove: EX_cys__L_e_i Score: 0.774±0.006
Size: 43 Remove: EX_ile__L_e_i Score: 0.770±0.021
Size: 42 Remove: EX_gsn_e_i Score: 0.774±0.042
Size: 41 Remove: EX_glcr_e_i Score: 0.778±0.020
Size: 40 Remove: EX_cit_e_i Score: 0.782±0.025
Size: 39 Remove: EX_acser_e_i Score: 0.782±0.012
Size: 38 Remove: EX_pppn_e_i Score: 0.790±0.010
Size: 37 Remove: EX_urea_e_i Score: 0.794±0.012
Size: 36 Remove: EX_taur_e_i Score: 0.794±0.015
Size: 35 Remove: EX_4abut_

ALL Size: 13 Method: svm_classifier Score: accuracy_score Acc: 0.823±0.015 for 20-fold-CV and 5 iter
Selected features:
'HILIC-M195' 'C18-M89' 'C18-M318' 'C18-M326' 'C18-M74' 'C18-M417'
 'C18-M142' 'HILIC-M98' 'C18-M305' 'HILIC-M113' 'C18-M347' 'C18-M388'
 'HILIC-M91'
ALL Size: 8 Method: XGB Score: accuracy_score Acc: 0.860±0.020 for 20-fold-CV and 5 iter
Selected features:
'C18-M387' 'HILIC-M195' 'C18-M326' 'C18-M74' 'C18-M142' 'C18-M62'
 'HILIC-M185' 'HILIC-M91
 ALL Size: 11 Method: MLP Score: accuracy_score Acc: 0.844±0.023 for 20-fold-CV and 5 iter
Selected features:
'C18-M89' 'C18-M318' 'C18-M326' 'C18-M417' 'C18-M142' 'C18-M377'
 'C18-M286' 'C18-M382' 'C18-M47' 'C18-M388' 'HILIC-M91'
